iati/core/rulesets.py

```diff
@@ -774,7 +774,7 @@ def is_valid_for(self, dataset):
                     unique.add(string_to_check)
             if len(original) != len(unique):
                 return False
-            original.clear()
+            original = list()  # Python 2 does not have a `list.clear()` function
```

Comment:
this isn't clearing a list, it's just assigning a new empty list. A better fix would be using del or slice.

https://stackoverflow.com/questions/1400608/how-to-empty-a-list-in-python

Reply:
Since the list values aren't used on the next iteration or after the loop, it's functionally the same thing

Reply:
it's slower and isn't as intuitive to read